### PR TITLE
added arbitrary item sorting

### PR DESCRIPTION
**Pull request by K-91**

### CODE REVIEW

**Comment by K-91** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ModComponents/ItemSorting.cs [hunk 1]
```diff
@@ -192,11 +191,40 @@ public static List<ItemIndex> sortItems(List<ItemIndex> itemList, Inventory inve
             List<SortStep> steps = new List<ItemSorting.SortStep>();
             bool filtering = false;
             bool tierSelect = false;
-            bool tierReversed = false;
+            bool sortReversed = false;
+            bool nameParsing = false;
+            StringBuilder parsedName = new StringBuilder();
```
Comment (xoxfaby) [inline ModComponents/ItemSorting.cs:196]: don't create a string builder every time, use the BetterUI shared stringbuilder

File: ModComponents/ItemSorting.cs [hunk 2]
```diff
@@ -143,7 +142,7 @@ public static List<EquipmentIndex> sortItems(List<EquipmentIndex> equipmentList,
         };
 
 
-        private static IOrderedEnumerable<ItemIndex> filteredSort<T>(IOrderedEnumerable<ItemIndex> list, Inventory inventory, ItemSorter<T> sorter, ItemFilter filter = null, bool reversed = false)
+        private static IOrderedEnumerable<ItemIndex> filteredSort<T>(IOrderedEnumerable<ItemIndex> list, Inventory inventory, ItemSorter<T> sorter, ItemFilter filter = null, bool reversed = false, ItemIndex idx = ItemIndex.None)
```
Comment (xoxfaby) [inline ModComponents/ItemSorting.cs:145]: lol

**Review by xoxfaby — CHANGES_REQUESTED** [review]
:)

**Review by xoxfaby — APPROVED** [review]

**Pull request merged**